fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -102,41 +102,40 @@ def __init__(
-        self.sharded_optimizers = [sharded_optimizer] if not isinstance(sharded_optimizer, list) else sharded_optimizer
-        self.enable_broadcast_buffers = broadcast_buffers
-        self.auto_refresh_trainable = auto_refresh_trainable
-        self.reduce_fp16 = reduce_fp16
+        self._module = module
+        self._sharded_optimizers = [sharded_optimizer] if not isinstance(sharded_optimizer, list) else sharded_optimizer
+        self._enable_broadcast_buffers = broadcast_buffers
+        self._auto_refresh_trainable = auto_refresh_trainable
+        self._reduce_fp16 = reduce_fp16
         if reduce_buffer_size > 0 and reduce_fp16:
-            self.reduce_fp16 = False
+            self._reduce_fp16 = False
             logging.warning(
                 "fp16 gradient reduction is not compatible with reduction buffers, which are requested. fp16 grad reduction is deactivated."
             )
 
         # Handle a no_sync() context which prevents the gradient synchronization,
         # accumulate in place
-        self.should_accumulate_grads = False
-        self.accumulate_grads_flipped = False
+        self._should_accumulate_grads = False
+        self._accumulate_grads_flipped = False
 
         # Communication related attributes
-        self.process_group = process_group if process_group is not None else dist.group.WORLD
-        self.backend = dist.get_backend(self.process_group)
-        self.world_size_scaling = 1.0 / dist.get_world_size(self.process_group)  # > 0
-        self.reference_global_rank = get_global_rank(self.process_group, 0)  # picking rank 0 as the reference
-        self.rank = dist.get_rank(self.process_group)
-        self.global_rank = get_global_rank(self.process_group, self.rank)
+        self._process_group = process_group if process_group is not None else dist.group.WORLD
+        self._backend = dist.get_backend(self._process_group)
+        self._world_size_scaling = 1.0 / dist.get_world_size(self._process_group)  # > 0
+        self._reference_global_rank = get_global_rank(self._process_group, 0)  # picking rank 0 as the reference
+        self._rank = dist.get_rank(self._process_group)
+        self._global_rank = get_global_rank(self._process_group, self._rank)
         self._local_to_global_rank = [
-            get_global_rank(self.process_group, i) for i in range(dist.get_world_size(self.process_group))
+            get_global_rank(self._process_group, i) for i in range(dist.get_world_size(self._process_group))
         ]
 
         # Expose some of the PytorchDDP attributes, some frameworks rely on them.
         # See https://pytorch.org/docs/stable/_modules/torch/nn/parallel/distributed.html#DistributedDataParallel
         # device_id related logic is not present, this is not handled
-        devices = {p.device for p in self.module.parameters()}
+        devices = {p.device for p in self._module.parameters()}
         self.is_multi_device_module = len(devices) > 1
```

Comment:
this is explicitly exposed because DDP exposes it, and some frameworks use it as a consequence